pkg/mic/mic.go

```diff
@@ -815,10 +816,16 @@ func getIDKey(ns, name string) string {
 	return strings.Join([]string{ns, name}, "/")
 }
 
-func (c *Client) convertIDListToMap(arr []aadpodid.AzureIdentity) (m map[string]aadpodid.AzureIdentity, err error) {
-	m = make(map[string]aadpodid.AzureIdentity, len(arr))
-	for _, element := range arr {
-		m[getIDKey(element.Namespace, element.Name)] = element
+func (c *Client) convertIDListToMap(azureIdentities []aadpodid.AzureIdentity) (m map[string]aadpodid.AzureIdentity, err error) {
+	m = make(map[string]aadpodid.AzureIdentity, len(azureIdentities))
+	for _, azureIdentity := range azureIdentities {
+		// validate the resourceID in azure identity to ensure format is as expected
+		err := utils.ValidateResourceID(azureIdentity.Spec.ResourceID)
+		if err != nil {
+			klog.Errorf("Ignoring azure identity %s/%s, error: %v", azureIdentity.Namespace, azureIdentity.Name, err)
```

Comment:
Should this be a warning ?

Reply:
I think we should keep it as an error. If users are filtering logs based on errors, this should surface out as to why the pod didn't get an `AzureAssignedIdentity`.